File: java/20210208/src/Ex05.java

public class Ex05 {

		int a = 10;
		
		public void doA() {
			a++;
		}
		
		public void doB() {
			a=20;
		}
		
		public static void main(String[] args) {
			
			Ex05 e5 = new Ex05();
			//e5.main(null);
			System.out.println("e5.a="+e5.a);
			//ex05를 인스턴스화 함.
			e5.doA();
			System.out.println("e5.a="+e5.a);
			e5.doB();
			System.out.println("e5.a="+e5.a);
		}
}

File: c_work21/20210216/ex05.c
#include <stdio.h>

typedef struct{
    int xpos;
    int ypos;
} Point;

//원점 대칭함수
void doA(Point *ptr){
    ptr->xpos = (ptr->xpos)*-1;
    ptr->ypos = (ptr->ypos)*-1;
}

//출력함수
void doB(Point pos){
    printf("pos.xpos = %d, pos.ypos = %d",pos.xpos,pos.ypos);
    pos.xpos = 20;
    printf("doB-----------\n");
    printf("pos.xpos = %d, pos.ypos = %d",pos.xpos,pos.ypos);
}


void main(){
    Point pos = {10,-5};
    doA(&pos);
    doB(pos);
    printf("main-----------\n");
    printf("pos.xpos = %d, pos.ypos = %d",pos.xpos,pos.ypos);
}